bolt-modules/boltlib/lib/puppet/functions/get_targets.rb

```diff
@@ -3,6 +3,8 @@
 require 'bolt/error'
 
 # Parses common ways of referring to targets and returns an array of Targets.
+#
+# **NOTE:** Not available in apply block if using Inventory version 2
```

Comment:
In every other case where we say "not available in apply block" an error is raised when the offending function is called in an apply block. In this case we just have empty data. See https://github.com/puppetlabs/bolt/blob/600252ce36bf624b7a53c9365e66e23b67829cd4/lib/bolt/inventory/inventory2.rb#L106-L116 . Unfortunately I think there is a bug. For example `get_targets('all')` returns an empty set of targets. But `get_targets('foo')` raises a random error. 
```
ab8omb37vvw9i8f.delivery.puppetlabs.net: Evaluation Error: Error while evaluating a Function Call, undefined method `[]' for nil:NilClass
```
I think for now the desired behavior is to always return an empty result list? It is certainly not to raise that exception.

Reply:
https://github.com/puppetlabs/bolt/pull/1105 is a proposed fix. Still needs tests if we decide this is the desired behavior.

Reply:
Updated with tests.